File: ruby/0253-meeting-rooms-ii.rb
def min_meeting_rooms(intervals)

   start = intervals.collect(&:first).sort!
   close = intervals.collect(&:last).sort!
    
   res = 0
   count = 0
   s = 0
   e = 0
    
   while (s < intervals.length)
    if(start[s] >= close[e])
     e +=1
     count -=1
    else
     s +=1
     count +=1
    end
    res = [res,count].max
   end
    return res
end

File: swift/0253-meeting-rooms-ii.swift
class Solution {
    func minMeetingRooms(_ intervals: [[Int]]) -> Int {
        var start = intervals.map({ $0[0] }).sorted()
        var end = intervals.map({ $0[1] }).sorted()
        var res = 0
        var count = 0
        var s = 0
        var e = 0

        while s < intervals.count {
            if start[s] < end[e] {
                s += 1
                count += 1
            } else {
                e += 1
                count -= 1
            }
            res = max(res, count)
        }

        return res
    }
}